### PR TITLE
Stabilize `duration_abs_diff`

### LINKED ISSUE #117618: Tracking Issue for `duration_abs_diff`
Feature gate: `#![feature(duration_abs_diff)]`

This is a tracking issue for the `Duration::abs_diff` method.

ACP: https://github.com/rust-lang/libs-team/issues/291

I haven't contributed to Rust before so I hope this is the right approach.

### Public API

```rust
impl Duration {
    pub const fn abs_diff(self, other: Duration) -> Duration;
}
```

### Steps / History

- [x] Implementation: #117619
- [x] Final comment period (FCP)[^1]
- [x] Stabilization PR: #127128

### Unresolved Questions

- None yet.

[^1]: https://std-dev-guide.rust-lang.org/feature-lifecycle/stabilization.html

### PR DESCRIPTION
**Pull request by elomatreb**
Stabilize `duration_abs_diff` following FCP in #117618. Closes #117618.

### CODE REVIEW

File: library/core/src/time.rs [hunk 1]
```diff
@@ -620,13 +620,12 @@ impl Duration {
...
     /// use std::time::Duration;
     ///
     /// assert_eq!(Duration::new(100, 0).abs_diff(Duration::new(80, 0)), Duration::new(20, 0));
     /// assert_eq!(Duration::new(100, 400_000_000).abs_diff(Duration::new(110, 0)), Duration::new(9, 600_000_000));
     /// ```
-    #[unstable(feature = "duration_abs_diff", issue = "117618")]
+    #[stable(feature = "duration_abs_diff", since = "CURRENT_RUSTC_VERSION")]
     #[must_use = "this returns the result of the operation, \
                   without modifying the original"]
     #[inline]
```
Comment (Nilstrieb) [inline library/core/src/time.rs:631]: should be entirely uncontroversial, but @rust-lang/wg-const-eval this will be const

**Comment by elomatreb** [comment]
I missed the bit in the [guide](https://std-dev-guide.rust-lang.org/development/stabilization.html) that told me to ping wg-const-eval when I initially made the PR and the tracking issue, sorry.

Since the tracking issue included the const-ness I added the attributes I needed to make it immediately stable and build, please tell me if there are mistakes.

File: library/core/src/time.rs (hunk 1 above)
Comment (RalfJung) [reply library/core/src/time.rs:633]: `rustc_allow_const_fn_unstable` is fishy. But I guess worst-case we can replace the `unwrap` by a custom `match` if we somehow need to preserve const-ness here and const_precise_live_drops does not work out...

Comment (Nilstrieb) [reply library/core/src/time.rs:633]: we've done worse crimes than inlining `Option::unwrap`, sounds fine :D

Comment (elomatreb) [reply library/core/src/time.rs:633]: > rustc_allow_const_fn_unstable is fishy.

I got that impression as well, I didn't expect having to add this. Should I get rid of this in favor of a match?

Comment (Nilstrieb) [reply library/core/src/time.rs:633]: I think it's ok to keep as-is, it's just a convenience that could be removed in the future if necessary, but we can keep while we can.

**Comment by joboet** [comment]
Seems like all the relevant people approve.
Congratulations!
@bors r+ rollup

**Pull request merged**